pkg/common/cns-lib/volume/manager.go

```diff
@@ -173,12 +176,21 @@ func (m *defaultManager) CreateVolume(ctx context.Context, spec *cnstypes.CnsVol
 	cnsCreateSpecList = append(cnsCreateSpecList, *spec)
 	var task *object.Task
 	var taskInfo *vim25types.TaskInfo
+	// store the volume name passed in by input spec, this name may exceed 80 characters
+	volNameFromInputSpec := spec.Name
 	// Call the CNS CreateVolume
-	taskDetailsInMap, ok := volumeTaskMap[spec.Name]
+	taskDetailsInMap, ok := volumeTaskMap[volNameFromInputSpec]
 	if ok {
 		task = taskDetailsInMap.task
-		log.Infof("CreateVolume task still pending for VolumeName: %q, with taskInfo: %+v", spec.Name, task)
+		log.Infof("CreateVolume task still pending for VolumeName: %q, with taskInfo: %+v", volNameFromInputSpec, task)
 	} else {
+		// truncate the volume name to make sure the name is within 80 characters before calling CNS
+		if len(cnsCreateSpecList[0].Name) > maxLengthOfVolumeNameInCNS {
+			volNameAfterTruncate := cnsCreateSpecList[0].Name[0 : maxLengthOfVolumeNameInCNS-1]
```

Comment:
nit: Can we truncate the name in spec instance itself, and then append it to cnsCreateSpecList? Will be better in terms of readability IMO.

Reply:
we need to make sure we do not use truncated value as the key to save tasks in the map, otherwise, there is a chance we will override the task of some other volume.

Reply:
only use the truncated value in the create spec while making create a call to VC, rest all places use non-truncated value.